dvc/tree/repo.py

```diff
@@ -306,6 +306,17 @@ def walk_files(self, top, **kwargs):  # pylint: disable=arguments-differ
             for fname in files:
                 yield PathInfo(root) / fname
 
+    def get_dir_hash(self, path_info, **kwargs):
+        if not self.exists(path_info):
+            raise FileNotFoundError
```

Comment:
We really need to start unifying exceptions in trees so we could use exception flow control instead of abusing `self.exists`.